tests/test_flow/test_copy.py

```diff
@@ -0,0 +1,138 @@
+    dir_path.mkdir()
+    flow.get('f', mode='FileCopier').copy(destination=dir_path)
+
+    expected_file_path = dir_path / 'f.pkl'
+    assert pickle.loads(expected_file_path.read_bytes()) == 5
+
+
+def test_copy_file_to_local_file(flow, tmp_path):
+    file_path = tmp_path / 'data.pkl'
+    flow.get('f', mode='FileCopier').copy(destination=file_path)
+
+    assert pickle.loads(file_path.read_bytes()) == 5
+
+
+def test_copy_file_to_local_file_using_str(flow, tmp_path):
+    file_path = tmp_path / 'data.pkl'
+    file_path_str = str(file_path)
+    flow.get('f', mode='FileCopier').copy(destination=file_path_str)
+    assert pickle.loads(file_path.read_bytes()) == 5
+
+
+@skip_unless_gcs
+def test_copy_file_to_gcs_dir(flow, tmp_path):
+    flow.get('f', mode='FileCopier').copy(destination='gs://' + GCS_TEST_BUCKET)
+    cloud_path = Path(GCS_TEST_BUCKET) / 'f.pkl'
+    local_path = tmp_path / 'f.pkl'
+    check_call('gsutil -m cp gs://{} {}'.format(cloud_path, local_path), shell=True)
+    assert pickle.loads(local_path.read_bytes()) == 5
+    check_call('gsutil -m rm gs://{}'.format(cloud_path), shell=True)
+
+
+@skip_unless_gcs
+def test_copy_file_to_gcs_file(flow, tmp_path):
+    cloud_path = str(Path(GCS_TEST_BUCKET) / 'f.pkl')
+    flow.get('f', mode='FileCopier').copy(destination='gs://' + cloud_path)
+    local_path = tmp_path / 'f.pkl'
+    check_call('gsutil -m cp gs://{} {}'.format(cloud_path, local_path), shell=True)
+    assert pickle.loads(local_path.read_bytes()) == 5
+    check_call('gsutil -m rm gs://{}'.format(cloud_path), shell=True)
+
+
+@dask_py3_only
+def test_copy_dask_to_dir(tmp_path, expected_dask_df, dask_flow):
+    destination = tmp_path / 'output'
+    destination.mkdir()
+    expected_dir_path = destination / 'dask_df.pq.dask'
+
+    dask_flow.get('dask_df', mode='FileCopier').copy(destination=destination)
+
+    actual = dd.read_parquet(expected_dir_path)
+    assert equal_frame_and_index_content(actual.compute(), expected_dask_df.compute())
+
+
+@skip_unless_gcs
+def test_copy_dask_to_gcs_dir(tmp_path, expected_dask_df, dask_flow):
+
+    cloud_path = str(Path(GCS_TEST_BUCKET) / 'output')
+    dask_flow.get('dask_df', mode='FileCopier').copy(destination='gs://' + cloud_path)
+
+    check_call('gsutil -m cp -r gs://{} {}'.format(cloud_path, tmp_path), shell=True)
+    actual = dd.read_parquet(tmp_path / 'output')
+    assert equal_frame_and_index_content(actual.compute(), expected_dask_df.compute())
+    check_call('gsutil -m rm -r gs://{}'.format(cloud_path), shell=True)
+
+
+def test_get_multi_value_entity(builder):
+
+    my_set = {"oscar", "the", "grouch"}
+    builder.assign('val', values=my_set)
+
+    @builder
+    def multi_entity(val):
+        return val
+
+    flow = builder.build()
+    results = flow.get('multi_entity', collection=set, mode=Path)
+    results = {pickle.loads(res.read_bytes()) for res in results}
+
+    assert results == my_set, "Expected {}, got {}".format(results, my_set)
```

Comment:
I'm not sure you need the explanatory string for this assert -- I think pytest will construct a useful message here automatically.